hibernate-core/src/test/java/org/hibernate/test/annotations/manytoone/NotNullManyToOneTest.java

```diff
@@ -0,0 +1,93 @@
+package org.hibernate.test.annotations.manytoone;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.cfg.beanvalidation.ValidationMode;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+/**
+ * @author Andrea Boriero
+ */
+@TestForIssue(jiraKey = "HHH-13959")
+public class NotNullManyToOneTest extends BaseCoreFunctionalTestCase {
+
+	@Override
+	protected void configure(Configuration configuration) {
+		configuration.setProperty( AvailableSettings.JPA_VALIDATION_MODE, ValidationMode.AUTO.name() );
+	}
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[] {
+				Parent.class,
+				Child.class
+		};
+	}
+
+	@Test
+	public void testSave() {
+		inTransaction(
+				session -> {
+					Parent parent = new Parent( new Child() );
+					session.save( parent );
+				}
+		);
+	}
+
+	@Test(expected = javax.validation.ConstraintViolationException.class)
+	public void testSaveWithNullChild() {
```

Comment:
the method name is a little bit confusing for the real issue is null `parent`

Reply:
fixed it, Thanks